### PR TITLE
Integrate notification to kibana-reports

### PR DESCRIPTION
**Pull request by zhongnansu**
*Issue #, if available:*

*Description of changes:*
- UI
- Server
  - schema
  - API

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by anirudha — APPROVED** [review]

**Review by abbashus — APPROVED** [review]

File: kibana-reports/server/backend/opendistro-notificaiton-plugin.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
```
Comment (davidcui-amzn) [inline kibana-reports/server/backend/opendistro-notificaiton-plugin.ts:1]: np: this file has a type: `notificaiton` vs `notification`

File: kibana-reports/server/routes/utils/__tests__/savedSearchReportHelper.test.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ import { reportSchema } from '../../../model';
  */
 const input = {
   query_url:
-    'http://localhost:5601/app/kibana#/search/7adfa750-4c81-11e8-b3d7-01146121b73d',
+    'http://localhost:5601/app/discover#/view/7adfa750-4c81-11e8-b3d7-01146121b73d',
```
Comment (davidcui-amzn) [inline kibana-reports/server/routes/utils/__tests__/savedSearchReportHelper.test.ts:25]: Does this PR include a version change? Why are we changing the URL?

**Review by davidcui-amzn — COMMENTED** [review]
Please address comments before merging

File: kibana-reports/server/routes/utils/__tests__/savedSearchReportHelper.test.ts (hunk 2 above)
Comment (zhongnansu) [reply kibana-reports/server/routes/utils/__tests__/savedSearchReportHelper.test.ts:25]: That was a typo, it should be discover view. But anyway, this field doesn't affect the test

File: kibana-reports/server/backend/opendistro-notificaiton-plugin.ts (hunk 1 above)
Comment (zhongnansu) [reply kibana-reports/server/backend/opendistro-notificaiton-plugin.ts:1]: good catch

**Pull request merged**